lit/lit
# [infra] update benchmark generator

### Context

For easier and more robust benchmarking, the generator is a fantastic tool for changing the parameters of template-heavy.

This change updates some old dependencies and ensures that the generator works.

### Fix

Some dependency updates, tsconfig update, and added a license to the generated output.

### How to test

Check out the branch. `git checkout origin/update-benchmark-generator`.

Then:

```sh
$ cd packages/benchmarks/generator/
$ ./scripts/generate.sh
Generating lit-html/template-heavy
Generating variant template-heavy
Done.
```

Note that `template-heavy.html` and `template-heavy.js` has been updated.


### Test plan

Also tested manually with small options to generate a tiny file. E.g. by setting width and depth to 1.

```sh
node generator/build/index.js  -r lit-html,lit-html@tot=generator/scripts/package-versions-tot.json,lit-html@release=generator/scripts/package-versions-release.json -n template-heavy -o lit-html/template-heavy -u 10 -q -m render,update -g -w 1 -d 1
```

which generated:

```ts
/**
 * @license
 * Copyright 2023 Google LLC
 * SPDX-License-Identifier: BSD-3-Clause
 */

/*
Benchmark generated via the following invocation:
node generator/build/index.js -r lit-html,lit-html@tot=generator/scripts/package-versions-tot.json,lit-html@release=generator/scripts/package-versions-release.json -n template-heavy -o lit-html/template-heavy -u 10 -q -m render,update -g -w 1 -d 1

Parameters:
  renderers: lit-html
  depth: 1
  width: 1
  attrPerNode: 4
  dynAttrPct: 0.5
  valPerDynAttr: 2
  dynNodePct: 0.5
  updateCount: 10
  updateNodePct: 1
  updateAttrPct: 1
  uniqueTemplates: true
  measure: render,update
*/

  import {html, render as litRender} from 'lit-html';
  
const render = () => html`<div a0_0="0_0" a0_1="0_1" a0_2="${'0_2:0#' + updateId}-${'0_2:1#' + updateId}" a0_3="${'0_3:0#' + updateId}-${'0_3:1#' + updateId}">0</div>`;

const container = document.getElementById('container');
let updateId = 0;
let useTemplateA = true;
window.update = () => {
  litRender(render(), container);
  updateId++;
  useTemplateA = !useTemplateA;
}
performance.mark('initial-render-start');
window.update();
performance.mark('initial-render-end');
for (let i=0; i<10; i++) {
  window.update();
}
performance.mark('updates-end');
performance.measure('render', 'initial-render-start', 'initial-render-end');
performance.measure('update', 'initial-render-end', 'updates-end');
performance.measure('time', 'initial-render-start', 'updates-end');
window.tachometerResult = performance.getEntriesByName('render')[0].duration +
      performance.getEntriesByName('update')[0].duration;
document.title = window.tachometerResult.toFixed(2) + 'ms';
```
